### PR TITLE
[PAY-829][PAY-1044] Desktop: Chats search users modal permissions/blocks and profile page permissions/blocks

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

- Gray out profiles you can't message on the search user modal
- Hides message icon from people you can message on their profile (TEMPORARY until we get CTA)
- Binds chat inbox settings to existing data

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/web/src/components/inbox-settings-modal/InboxSettingsModal.tsx [hunk 1]
```diff
@@ -98,6 +115,22 @@ export const InboxSettingsModal = () => {
...
 
+  // Fetch the latest permissions for the current user when the modal is made visible
+  // Note that this will trigger the following effect as well, causing the permission state to update
+  useEffect(() => {
+    if (isVisible && userId) {
+      dispatch(fetchPermissions({ userIds: [userId] }))
+    }
+  }, [dispatch, isVisible, userId])
+
+  // Since the modal is always mounted (booo!) we have to manually set the default
```
Comment (dharit-tan) [inline packages/web/src/components/inbox-settings-modal/InboxSettingsModal.tsx:126]: lol

File: packages/web/src/pages/profile-page/ProfilePageProvider.tsx [hunk 2]
```diff
@@ -828,6 +828,12 @@ class ProfilePage extends PureComponent<ProfilePageProps, ProfilePageState> {
       activeTab === ProfilePageTabs.REPOSTS ||
       activeTab === ProfilePageTabs.COLLECTIBLES
     const following = !!profile && profile.does_current_user_follow
+    const hasChatPermission =
+      (this.props.profile.profile?.user_id &&
+        !this.props.blockerList.includes(this.props.profile.profile.user_id) &&
+        this.props.permissionsMap[this.props.profile.profile.user_id]
+          ?.current_user_has_permission) ??
+      false
```
Comment (dharit-tan) [inline packages/web/src/pages/profile-page/ProfilePageProvider.tsx:836]: Can we remove the `?? false`?

**Review by dharit-tan — APPROVED** [review]

File: packages/common/src/store/pages/chat/sagas.ts [hunk 3]
```diff
@@ -278,6 +287,23 @@ function* doFetchBlockees() {
...
+    const { data } = yield* call([sdk.chats, sdk.chats.getBlockers])
+    yield* put(
+      fetchBlockersSucceeded({
+        blockers: data
+          .map((encodedId) => decodeHashId(encodedId))
+          .filter(removeNullable)
+      })
+    )
+  } catch (e) {
+    console.error('fetchBlockeesFailed', e)
```
Comment (piazzatron) [inline packages/common/src/store/pages/chat/sagas.ts:303]: nit: `fetchBlockersFailed`

File: packages/web/src/pages/chat-page/components/CreateChatUserResult.tsx [hunk 4]
```diff
@@ -66,7 +69,13 @@ export const MessageUserSearchResult = (props: UserResultComposeProps) => {
   const supportingMap = useSelector(getOptimisticSupporting)
   const supportersMap = useSelector(getOptimisticSupporters)
   const blockeeList = useSelector(getBlockees)
-  const isBlocked = blockeeList.includes(user.user_id)
+  const blockerList = useSelector(getBlockers)
+  const isBlockee = blockeeList.includes(user.user_id)
+  const isBlocker = blockerList.includes(user.user_id)
+  const permissionsMap = useSelector(getPermissionsMap)
+  const isPermitted =
+    !isBlocker &&
```
Comment (piazzatron) [inline packages/web/src/pages/chat-page/components/CreateChatUserResult.tsx:77]: should we also check `isBlockee` here (can't message ppl you've blocked)?

File: packages/web/src/pages/profile-page/ProfilePageProvider.tsx (hunk 2 above)
Comment (piazzatron) [reply packages/web/src/pages/profile-page/ProfilePageProvider.tsx:836]: that would make one codepath return undefined as the type here

**Pull request merged**